src/main/java/org/littleshoot/proxy/BadGatewayFailureHttpResponseComposer.java

```diff
@@ -0,0 +1,42 @@
+package org.littleshoot.proxy;
+
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+import org.littleshoot.proxy.impl.ProxyUtils;
+
+public class BadGatewayFailureHttpResponseComposer implements FailureHttpResponseComposer {
+
+  /**
+   * Tells the client that something went wrong trying to proxy its request. If the Bad Gateway is a response to
+   * an HTTP HEAD request, the response will contain no body, but the Content-Length header will be set to the
+   * value it would have been if this 502 Bad Gateway were in response to a GET.
+   *
+   * @param httpRequest the HttpRequest that is resulting in the Bad Gateway response
+   * @param cause raised exception
+   * @return true if the connection will be kept open, or false if it will be disconnected
+   */
+  @Override
+  public FullHttpResponse compose(HttpRequest httpRequest, Throwable cause) {
+    String body = provideCustomMessage(httpRequest, cause);
+
+    FullHttpResponse response = ProxyUtils.createFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.BAD_GATEWAY, body);
+
+    if (ProxyUtils.isHEAD(httpRequest)) {
```

Comment:
![badgatewaytest](https://cloud.githubusercontent.com/assets/18006133/26680575/a2540b0a-46e2-11e7-9539-f2d2460b802b.jpg)
well, it is tested by the original littleproxy tests.. but okay, lets add additional test
@osklyarenko